### PR TITLE
docs: add details in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by EuphoricThinking**
Add black installation details in CONTRIBUTING.md
as in README.md

### Description

Added information about `black` installation in CONTRIBUTING.md, as it is in README.md. A possible contributor (who could focus on CONTRIBUTING.md) might have problems with formatting, since targets `format-apply` and `format-check` are not built if `black` is not installed (issue based on a true story).
### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

**Review by KFilipek — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ source files, respectively.
 **NOTE**: We use specific versions of formatting tools to ensure consistency across the project. The required versions are:
 - clang-format version **15.0**, which can be installed with the command: `python -m pip install clang-format==15.0.7`.
 - cmake-format version **0.6**, which can be installed with the command: `python -m pip install cmake-format==0.6.13`.
+- black version **24.3**, which can be installed with the command: `python -m pip install black==24.3.0`.
```
Comment (lukaszstolarczuk) [inline CONTRIBUTING.md:100]: hmm.. this section is about `specific required versions` of these tools. We don't require a specific version of `black` (it has rather good compatibility between versions (in comparison to clang-format, which may change too much between versions).

So... I would not add this info here...?


//
General information how to install black (including suggested version) is in the top-level Readme - https://github.com/oneapi-src/unified-memory-framework?tab=readme-ov-file#build

**Review by lukaszstolarczuk — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (EuphoricThinking) [reply CONTRIBUTING.md]: Do you wish to leave the suggested version number? Simply `python -m pip install black` should suffice.

I have removed the version number in the updated version.

Comment (EuphoricThinking) [reply CONTRIBUTING.md]: Or maybe you mean that you don't want to leave the info about black at all?

Comment (lukaszstolarczuk) [reply CONTRIBUTING.md]: I mean this change does not introduce anything new in here, I believe. The same info is in the top-level README.

This section, useful for contributors, says `The required versions are: ...` and `black` does not have any specific version required (at least at the moment).

**Review by lplewa — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (lukaszstolarczuk) [reply CONTRIBUTING.md]: ok, I see your point now. Then perhaps we could write something like this:

```
- black - no specific version required (can be installed with the command: `python -m pip install black==24.3.0)`.
```

Comment (EuphoricThinking) [reply CONTRIBUTING.md]: Done

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**